Fix downloading of books whose title has trailing whitespace

eunikolsky opened the pull request:
This fixes the "Unchecked lastError value: Error: filename must not
contain illegal characters" error while trying to download an audiobook
whose title has a space at the end
(https://akniga.org/valeriya-volchya-semya-bruts). The generated
filenames had the following pattern:

```
Волчья Валерия - Семья Брутс /cover.jpg
Волчья Валерия - Семья Брутс /desc.txt
```

which are fine (verified in OS X 10.14.6), but Firefox 84.0.2 complained
about them.

~~The fix is to trim `path` after it's constructed from the `author` and
`title` strings so that it doesn't have leading or trailing whitespace.~~

Update: The fix is to trim author and title rather than the download directory.
This is done in the parser and also fixes a possible case with
leading/trailing whitespace, which could result in the following
undesirable path: `Author    -    Title/cover.jpg`. This commit removes
trimming of the download directory because that is no longer necessary
with the cleaned up parsed data.

eunikolsky committed:
Fix downloading of books whose title has trailing whitespace

This fixes the "Unchecked lastError value: Error: filename must not
contain illegal characters" error while trying to download an audiobook
whose title has a space at the end
(https://akniga.org/valeriya-volchya-semya-bruts). The generated
filenames had the following pattern:

```
Волчья Валерия - Семья Брутс /cover.jpg
Волчья Валерия - Семья Брутс /desc.txt
```

which are fine (verified in OS X 10.14.6), but Firefox 84.0.2 complained
about them.

The fix is to trim `path` after it's constructed from the `author` and
`title` strings so that it doesn't have leading or trailing whitespace.

phoenixrvd commented:
Thanks for the fix.

1. i would trim aer title and author separately to avoid following case:

```
Волчья Валерия    -     Семья Брутс /cover.jpg
Волчья Валерия    -     Семья Брутс /desc.txt
```

2. i would do it in the parser, because the capture of data is in responsibility of the parser. This will automatically fix the problem with spaces in desc.json content as well.

eunikolsky committed:
Trim author and title rather than the download directory

This is done in the parser and also fixes a possible case with
leading/trailing whitespace, which could result in the following
undesirable path: `Author    -    Title/cover.jpg`. This commit removes
trimming of the download directory because that is no longer necessary
with the cleaned up parsed data.

eunikolsky commented:
@phoenixrvd, thanks for the feedback. I'd thought of doing it in the parser, but decided against it because I didn't know the reason for that trailing space and wanted to preserve the parsed data as is. However a path like
```
Волчья Валерия    -     Семья Брутс /cover.jpg
```
would be undesirable as you pointed out, so I've updated the fix to trim `author` and `title` in the parser. Please re-review.
